### PR TITLE
FSET-1387. Notify candidate of FSAC results - 1st pref. needs FSB.

### PR DESCRIPTION
**Pull request by vglushak**

### CODE REVIEW

File: app/services/assessmentcentre/AssessmentCentreToFsbOrOfferProgressionService.scala [hunk 1]
```diff
@@ -59,14 +77,27 @@ trait AssessmentCentreToFsbOrOfferProgressionService extends CurrentSchemeStatus
...
       )
     }
-
     updates.map(SerialUpdateResult.fromEither)
   }
-}
+
+  private def retrieveCandidateDetails(applicationId: String)(implicit hc: HeaderCarrier) = {
+    applicationRepo.find(applicationId).flatMap {
+      case Some(app) => authProviderClient.findByUserIds(Seq(app.userId))
```
Comment (benjstephenson) [inline app/services/assessmentcentre/AssessmentCentreToFsbOrOfferProgressionService.scala]: You can get these details from the application document, you don't need to add a dependency on the auth provider

Comment (vglushak) [reply app/services/assessmentcentre/AssessmentCentreToFsbOrOfferProgressionService.scala]: Actually I can't - there are no email there.

Comment (benjstephenson) [reply app/services/assessmentcentre/AssessmentCentreToFsbOrOfferProgressionService.scala]: candidate details collection

Comment (vglushak) [reply app/services/assessmentcentre/AssessmentCentreToFsbOrOfferProgressionService.scala]: ok, but that means I have to add dependency to contact-details repository, isn't it?

**Comment by benjstephenson** [comment]
True, but means that we're not calling another service outside of FS
backend.

File: app/services/assessmentcentre/AssessmentCentreToFsbOrOfferProgressionService.scala (hunk 1 above)
Comment (benjstephenson) [reply app/services/assessmentcentre/AssessmentCentreToFsbOrOfferProgressionService.scala]: True, but it means we don't have to call an external service

File: test/services/assessmentcentre/AssessmentCentreToFsbOrOfferProgressionServiceSpec.scala [hunk 2]
```diff
@@ -38,10 +41,22 @@ class AssessmentCentreToFsbOrOfferProgressionServiceSpec extends ScalaMockUnitSp
...
+          .expects(expectedApplication.applicationId)
+          .returningAsync(Option(c0)).once
+
+        (mockAuthProviderClient.findByUserIds(_: Seq[String])(_: HeaderCarrier))
+          .expects(Seq(userId), hc)
+          .returningAsync(Seq(c1)).once
+
+        (mockEmailClient.sendCandidateAssessmentCompletedMovedToFsb(_: String, _: String)(_: HeaderCarrier))
+          .expects(c1.email, c1.name, hc)
+          .returningAsync.once
```
Comment (benjstephenson) [inline test/services/assessmentcentre/AssessmentCentreToFsbOrOfferProgressionServiceSpec.scala]: I'm pretty sure that you can omit the .once calls and the default is that the method is only called once

**Comment by vglushak** [comment]
Both comments addressed!

**Comment by benjstephenson** [comment]
LGTM 👍  Will wait for Travis to be happy and then merge

**Pull request merged**